src/kernel.hpp

```diff
@@ -319,6 +329,22 @@ struct cvk_kernel_argument_values {
                 }
 
                 m_kernel_resources[arg.binding] = sampler;
+            } else if (arg.kind == kernel_argument_kind::buffer) {
+                auto device_ptr =
+                    *reinterpret_cast<const cl_mem_device_address_EXT*>(value);
+                auto device_to_buffer_map = m_entry_point->program()
+                                                ->context()
+                                                ->device()
+                                                ->device_to_buffer_map;
+                auto it = device_to_buffer_map.find((void*)device_ptr);
+                if (it == device_to_buffer_map.end()) {
+                    return CL_INVALID_MEM_OBJECT;
+                }
+
+                // device pointer found in map, swapping the buffer pointer
+                auto buffer_ptr_raw = it->second;
+                auto buffer_ptr = reinterpret_cast<cvk_buffer*>(buffer_ptr_raw);
+                m_kernel_resources[arg.binding] = buffer_ptr;
```

Comment:
This will break the existing argument setting for buffers. It might be easier/cleaner to introduce a new `cvk_kernel::set_arg_device_address` function (that would have to also mark the argument as set as done by line 361).